### PR TITLE
Free environment variable table in inprocess application

### PR DESCRIPTION
**Pull request by jkotalik**
Found using appverifier. Test will be enabled through adding app verifier to VSTS (#1244 attempted to do it, will enable once all failures are resolved).

### CODE REVIEW

File: src/AspNetCoreModuleV2/InProcessRequestHandler/inprocessapplication.cpp [hunk 1]
```diff
@@ -403,18 +403,19 @@ IN_PROCESS_APPLICATION::SetEnvironementVariablesOnWorkerProcess(
...
-    pHashTable->Apply(ENVIRONMENT_VAR_HELPERS::AppendEnvironmentVariables, &hr);
-    if (FAILED(hr))
-    {
-        goto Finished;
-    }
-    pHashTable->Apply(ENVIRONMENT_VAR_HELPERS::SetEnvironmentVariables, &hr);
+    table->Apply(ENVIRONMENT_VAR_HELPERS::AppendEnvironmentVariables, &hr);
     if (FAILED(hr))
     {
         goto Finished;
```
Comment (pakrym) [inline src/AspNetCoreModuleV2/InProcessRequestHandler/inprocessapplication.cpp]: macrosify

File: src/AspNetCoreModuleV2/InProcessRequestHandler/inprocessapplication.cpp [hunk 2]
```diff
@@ -403,18 +403,19 @@ IN_PROCESS_APPLICATION::SetEnvironementVariablesOnWorkerProcess(
...
     }
+
+    table->Apply(ENVIRONMENT_VAR_HELPERS::SetEnvironmentVariables, &hr);
+
 Finished:
+    table->Clear();
```
Comment (pakrym) [inline src/AspNetCoreModuleV2/InProcessRequestHandler/inprocessapplication.cpp]: You can add a deleter that would clear table too.

Comment (jkotalik) [reply src/AspNetCoreModuleV2/InProcessRequestHandler/inprocessapplication.cpp]: Yeah I'll do that.

**Comment by pakrym** [comment]
I don't like appveyor failure

**Comment by jkotalik** [comment]
Yeah that one seems concerning. I doubt it's related though?

**Comment by pakrym** [comment]
I've never seen it, and it might be caused by a  shutdown AV.

**Comment by jkotalik** [comment]
`AppOfflineDropped_CanRemoveAppOfflineAfterAddingAndSiteWorks` seems to be unrelated to this change. I'll file an issue.

**Comment by pakrym** [comment]
Rerun in too

**Pull request merged**